tool/tsh/common/kube_proxy_test.go

```diff
@@ -130,15 +137,115 @@ func sendRequestToKubeLocalProxy(t *testing.T, config *clientcmdapi.Config, tele
 		KeyData:    config.AuthInfos[contextName].ClientKeyData,
 		ServerName: common.KubeLocalProxySNI(teleportCluster, kubeCluster),
 	}
-
-	client, err := kubernetes.NewForConfig(&rest.Config{
+	restConfig := &rest.Config{
 		Host:            "https://" + teleportCluster,
 		TLSClientConfig: tlsClientConfig,
 		Proxy:           http.ProxyURL(proxyURL),
-	})
+	}
+	client, err := kubernetes.NewForConfig(restConfig)
 	require.NoError(t, err)
 
 	resp, err := client.CoreV1().Pods("default").List(context.Background(), metav1.ListOptions{})
 	require.Nil(t, err)
 	require.GreaterOrEqual(t, len(resp.Items), 1)
+
+	runKubectlExec(t, restConfig)
+}
+
+// runKubectlExec runs a kubectl exec command in a dummy pod.
+// The mock Kubernetes API server will return the pod name and the stdin data
+// written to the pod.
+func runKubectlExec(t *testing.T, config *rest.Config) {
```

Comment:
Is there a cheap way to replace [`mustGetKubePod`](https://github.com/gravitational/teleport/blob/31fd98a71d8ecf8548dd3823ae965392765874b8/integration/proxy/proxy_helpers.go#L745) in integration tests with doing kubectl exec? Or maybe the integration tests should do both?

Reply:
You mean downloading kubectl instead of running all this weird logic?

Reply:
No, I meant replicating what you did in tsh tests in this PR in integration tests for kube proxy. If integration tests also did exec into a container then maybe we'd have caught that sooner.

I typically also did just `kubectl get po -A` to verify that kube access in Connect works as expected, but I'm going to update the test plan to explicitly mention execing into a container.